php/libraries/ICandidate.class.inc

```diff
@@ -0,0 +1,319 @@
+
+    /**
+     * Manages the parameter_candidate tableso that projects have a way to add
+     * whatever data they want which is outside the scope of LORIS to their
+     * candidates.
+     *
+     * @param string|null $var The type of data about the Candidate
+     *                    to be returned
+     *
+     * @return     mixed The info requested
+     * @deprecated
+     */
+    public function getData(?string $var = null);
+
+
+    /**
+     * Returns the ProjectID for the current candidate
+     *
+     * @return integer
+     */
+    public function getProjectID(): int;
+
+    /**
+     * Returns the name of the candidate's project from the
+     * config file
+     *
+     * @return string Project name
+     */
+    public function getProjectTitle(): string;
+
+    /**
+     * Returns the CandID of this candidate
+     *
+     * @return CandID 6 digit CandID
+     */
+    public function getCandID();
+
+    /**
+     * Returns the PSCID of this candidate
+     *
+     * @return string The PSCID identifier for candidate
+     */
+    public function getPSCID(): string;
+
+    /**
+     * Returns the site of this candidate
+     *
+     * @return Site
+     */
+    public function getCandidateSite(): Site;
+
+
+    /**
+     * Returns the ID of this candidate's site
+     *
+     * @return int The numeric ID of the candidate's site
+     */
+    public function getCenterID(): int;
+
+    /**
+     * Returns the date of birth of this candidate
+     *
+     * @return ?DateTime
+     */
+    public function getCandidateDoB(): ?DateTime;
+
+    /**
+     * Returns the expect date of confinement (due date)
+     * of this candidate.
+     *
+     * @return ?DateTime
+     */
+    public function getCandidateEDC(): ?DateTime;
+
+    /**
+     * Return the candidate's sex
+     *
+     * @return string|null Male|Female String describing the sex
+     *                     and null if scanner
+     */
+    public function getCandidateSex(): ?string;
+
+    /**
+     * Return whether the candidate is active
+     *
+     * @return bool
+     */
+    public function isActive(): bool;
+
+    /**
+     * Return the user that registered (created) this candidate
+     *
+     * @return string UserID of user who created this candidate.
+     */
+    public function getRegisteredBy(): string;
+
+    /**
+     * Return the name of the user account that last changed this candidate.
+     *
+     * @return User
+     */
+    public function getLastModifiedBy(): User;
+
+    /**
+     * Getter method for entityType class variable.
+     *
+     * @return string The value of entityType. 'Human' or 'Scanner'.
+     */
+    public function getEntityType(): string;
+
+    /**
+     * Returns the list of visit labels w/ corresponding SessionID as key
+     *
+     * @return array<int,string> Associative array in the format
+     *                              SessionID=>Visit_label
+     */
+    public function getListOfVisitLabels(): array;
```

Comment:
That should be covered by `getTimepoints()`. 
but if it's really necessary, it should return an array of String; not an array of key-value pairs.